### ISSUE TITLE
Not able to instrument fn from REPL using the UI

### ISSUE THREAD

**Issue by ikappaki** [jpmonettas/flow-storm-debugger#70]
Hi,

I can't seem to be able to instrument a fn using the Browser->ns->fn->instrument button while interacting with the REPL. When I eval the fn, nothing shows up in the Flows tab in the UI.

To reproduce
1. Create an example namespaced fn in `src/issue.clj`
```clj
(ns issue)

(defn hello []
  (println :hi))
```
2.  Bring up the REPL
``` powershell 
clj -Sforce -Sdeps '{:deps {com.github.jpmonettas/flow-storm-dbg {:mvn/version """RELEASE"""}}}'
```
3. Bring up flow storm
``` clojure
(do (require '[flow-storm.api :as fs-api]) (fs-api/local-connect))
```
4. load example file
``` clojure
(load-file "src/issue.clj")
```
5. On the flow storm UI, instrument the hello fn
![image](https://user-images.githubusercontent.com/34983288/215318665-75a007f0-b6ea-46d4-8121-1fd7669ca203.png)
6. Back on the REPL, eval the fn
``` clojure
(require '[issue])
(issue/hello)
```
7. There's no trace in the `flow` tab
![image](https://user-images.githubusercontent.com/34983288/215318720-9dfff47d-50a7-4292-894b-ea7ef7d1a10e.png)

The expectation is that the `issue/hello` eval trace will appear in the `flow` tab.

The same work as expected if instead of going through the UI, there's a `#trace` before the fn in the example file.

There's a suspicious `Couldn't find source for issue/hello` msg in the repl output after clicking the `instrument` button in the UI:
```
Clojure 1.11.1
user=> (do (require '[flow-storm.api :as fs-api]) (fs-api/local-connect))
[Starting Events queue subsystem]
Events queue started
[Starting Runtime Api subsystem]
[Starting docs subsystem]
Loading docs using () files
[Starting UI subsystem]
Jan 29, 2023 com.sun.javafx.application.PlatformImpl startup
WARNING: Unsupported JavaFX configuration: classes were loaded from 'unnamed module @496a023a'
#object[flow_storm.runtime.taps$setup_tap_BANG_$tap_f__10647 0x7d043960 "flow_storm.runtime.taps$setup_tap_BANG_$tap_f__10647@7d043960"]
user=> (load-file "src/issue.clj")
#'issue/hello
user=> Instrumenting var issue/hello {}
Re-evaluating var: issue/hello deep?: null instrument?: true
Couldn't find source for issue/hello

user=> (require '[issue])
nil
user=> (issue/hello)
:hi
nil
```

Thanks

**Comment by jpmonettas** [comment]
Hi! Thanks for the detailed report!

When you reported I tested it with a fn like `clojure.set/difference` and wasn't seeing the issue, but it happened only for loaded files, not compiled ones.

It was a bug on hansel, and it was fixed here https://github.com/jpmonettas/hansel/commit/4435a53dbd8a451590e0da688f4d7af1633e0592

I just released a new version of FlowStorm `3.3.313` that uses the fixed Hansel version, so if you use that one you shouldn't see that issue.

There is one possible way of getting the exact same issue, and is when you eval a fn from the repl and you try to instrument it from the browser without saving the file first. It will see the fn in the environment but it will not be able to find the source since you haven't save the changes, but it is pretty rare and very easy to avoid, just by saving the files you are trying to instrument.

**Comment by ikappaki** [comment]
> It was a bug on hansel, and it was fixed here [jpmonettas/hansel@4435a53](https://github.com/jpmonettas/hansel/commit/4435a53dbd8a451590e0da688f4d7af1633e0592)
> 
> I just released a new version of FlowStorm `3.3.313` that uses the fixed Hansel version, so if you use that one you shouldn't see that issue.

Thanks! I can confirm the fix is working!
 
> There is one possible way of getting the exact same issue, and is when you eval a fn from the repl and you try to instrument it from the browser without saving the file first. It will see the fn in the environment but it will not be able to find the source since you haven't save the changes, but it is pretty rare and very easy to avoid, just by saving the files you are trying to instrument.

It's good to know :)
